frontend/apps/filemanager/filemanagerhistory.lua

```diff
@@ -171,50 +191,32 @@ function FileManagerHistory:onShowHist()
         _manager = self,
     }
 
+    self.filter = G_reader_settings:readSetting("history_filter", "all")
+    if self.filter ~= "all" then
+        self:getHistStatuses()
+    end
     self:updateItemTable()
     self.hist_menu.close_callback = function()
-        self.statuses_fetched = nil
-        self.filter = nil
         UIManager:close(self.hist_menu)
     end
     UIManager:show(self.hist_menu)
     return true
 end
 
 function FileManagerHistory:showHistDialog()
-    if not self.statuses_fetched then
-        local status
-        for _, v in ipairs(require("readhistory").hist) do
-            if v.dim then
-                status = "deleted"
-            else
-                if DocSettings:hasSidecarFile(v.file) then
-                    local docinfo = DocSettings:open(v.file) -- no io handles created, do not close
-                    if docinfo.data.summary and docinfo.data.summary.status
-                            and docinfo.data.summary.status ~= "" then
-                        status = docinfo.data.summary.status
-                    else
-                        status = "reading"
-                    end
-                else
-                    status = "new"
-                end
-            end
-            v.status = status
-            self.count[status] = self.count[status] + 1
-        end
-        self.statuses_fetched = true
+    if self.filter == "all" then
+        self:getHistStatuses()
     end
-
     local hist_dialog
     local buttons = {}
-    local function genFilterButton(status)
+    local function genFilterButton(filter)
         return {
-            text = T(_("%1 (%2)"), status_text[status], self.count[status]),
+            text = T(_("%1 (%2)"), filter_text[filter], self.count[filter]),
             callback = function()
                 UIManager:close(hist_dialog)
-                self.filter = status ~= "all" and status
+                self.filter = filter
                 self:updateItemTable()
+                G_reader_settings:saveSetting("history_filter", self.filter)
```

Comment:
You can do it once in `hist_menu.close_callback`, to avoid duplicating in each filter button.

Reply:
Excellent